### PR TITLE
Adding unit tests for local_test_cluster

### LINKED ISSUE #532: Add unit test coverage for test workflow
Improve the test_workflow UT coverage. 

Report as of 20 Sept. 

Module | statements | missing | excluded | coverage
-- | -- | -- | -- | --
Total | 330 | 198 | 0 | 40%
src/test_workflow/integ_test/integ_test_suite.py | 50 | 50 | 0 | 0%
src/test_workflow/integ_test/local_test_cluster.py | 95 | 95 | 0 | 0%
src/test_workflow/test_args.py | 30 | 30 | 0 | 0%
src/test_workflow/test_recorder.py | 9 | 9 | 0 | 0%
src/test_workflow/test_cluster.py | 18 | 8 | 0 | 56%
src/test_workflow/test_component.py | 9 | 3 | 0 | 67%
src/test_workflow/bwc_test/bwc_test_suite.py | 33 | 2 | 0 | 94%
src/test_workflow/perf_test/perf_test_suite.py | 21 | 1 | 0 | 95%
src/test_workflow/__init__.py | 0 | 0 | 0 | 100%
src/test_workflow/bwc_test/__init__.py | 0 | 0 | 0 | 100%
src/test_workflow/dependency_installer.py | 23 | 0 | 0 | 100%
src/test_workflow/integ_test/__init__.py | 0 | 0 | 0 | 100%
src/test_workflow/perf_test/__init__.py | 0 | 0 | 0 | 100%
src/test_workflow/perf_test/perf_test_cluster.py | 42 | 0 | 0 | 100%
src/test_workflow/utils/__init__.py | 0 | 0 | 0 | 100%

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha <vachshah@amazon.com>

### Description
Added unit tests for `local_test_cluster.py`. 
 
### Issues Resolved
#532 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from unittest.mock import MagicMock, mock_open, patch
+
+import requests
+
+from manifests.bundle_manifest import BundleManifest
+from test_workflow.integ_test.local_test_cluster import LocalTestCluster
+from test_workflow.test_cluster import ClusterCreationException
+
+
+class MockResponse:
```
Comment (dblock) [inline bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py:20]: Move under `LocalTestClusterTests`.

File: bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    def setUp(self):
+        self.maxDiff = None
+        self.data_path = os.path.realpath(
+            os.path.join(os.path.dirname(__file__), "../../tests_manifests/data")
+        )
+        self.manifest_filename = os.path.join(
+            self.data_path, "opensearch-bundle-1.1.0.yml"
+        )
+        self.manifest = BundleManifest.from_path(self.manifest_filename)
+        with tempfile.TemporaryDirectory() as work_dir:
```
Comment (dblock) [inline bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py:36]: This does out of scope after `with` and creates a new directory. If you just need a dummy directory name you can do `tempfile.gettempdir()`

File: bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            self.local_test_cluster = LocalTestCluster(work_dir, "index-management", "", self.manifest, True, "dummy-bucket")
+
+    def mocked_response(*args, **kwargs):
```
Comment (dblock) [inline bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py:39]: mocked -> mock

NBD, but make private `__mock_response__`

**Review by dblock — APPROVED** [review]

File: bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            False,
+            "dummy-bucket",
+        )
+        with self.assertRaises(ClusterCreationException):
+            local_test_cluster.wait_for_service()
+            requests.get.assert_called_once_with(
+                "http://localhost:9200/_cluster/health",
+                verify=False,
+                auth=("admin", "admin"),
+            )
```
Comment (dblock) [inline bundle-workflow/tests/test_integ_workflow/integ_test/test_local_test_cluster.py:132]: You can check the text in the exception, too.

**Review by saratvemulapalli — APPROVED** [review]
Looks good to me.
Thanks @VachaShah !

**Review by dblock — APPROVED** [review]

**Pull request merged**